src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/AddressRepositoryIT.java

```diff
@@ -165,6 +166,25 @@ public void canDeleteByPartitionedCity() {
         assertThat(result.get(0).getCity()).isNotEqualTo(TEST_ADDRESS1_PARTITION1.getCity());
     }
 
+    @Test(expected = CosmosDBAccessException.class)
+    public void testDeleteByIdAndPartitionKey() {
+        final long count = repository.count();
+        assertThat(count).isEqualTo(4);
+
+        final Optional<Address> addressById = repository.findById(TEST_ADDRESS1_PARTITION1.getPostalCode(),
+            new PartitionKey(TEST_ADDRESS1_PARTITION1.getCity()));
+        assertThat(addressById.isPresent()).isTrue();
+
+        repository.deleteById(TEST_ADDRESS1_PARTITION1.getPostalCode(),
+            new PartitionKey(TEST_ADDRESS1_PARTITION1.getCity()));
+
+        final List<Address> result = TestUtils.toList(repository.findAll());
+        assertThat(result.size()).isEqualTo(3);
+
+        repository.findById(TEST_ADDRESS1_PARTITION1.getPostalCode(),
+            new PartitionKey(TEST_ADDRESS1_PARTITION1.getCity()));
```

Comment:
No assert here?

Reply:
This throws CosmosDbAccessException, since we are trying to find the same entity which we just deleted.
That's why this method expects CosmosDbAccessException.
We need to update our Exceptions to be more specific.